e2e/testcases/hydration_test.go

```diff
@@ -278,61 +249,41 @@ func TestHydrateHelmOverlay(t *testing.T) {
 		nt.MustMergePatch(rs, `{"spec": {"git": {"dir": "helm-overlay"}}}`)
 	}
 
-	err := nt.WatchForAllSyncs(nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: "helm-overlay"}))
-	if err != nil {
-		nt.T.Fatal(err)
-	}
+	nt.Must(nt.WatchForAllSyncs(nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: "helm-overlay"})))
 
 	nt.T.Log("Validate resources are synced")
-	if err := nt.Validate("my-coredns-coredns", "coredns", &appsv1.Deployment{},
+	nt.Must(nt.Validate("my-coredns-coredns", "coredns", &appsv1.Deployment{},
 		testpredicates.HasAnnotation("hydration-tool", "kustomize"),
 		testpredicates.HasLabel("team", "coredns"),
 		testpredicates.HasAnnotation("client.lifecycle.config.k8s.io/mutation", "ignore"),
 		testpredicates.HasAnnotation(metadata.KustomizeOrigin, "configuredIn: base/kustomization.yaml\nconfiguredBy:\n  apiVersion: builtin\n  kind: HelmChartInflationGenerator\n"),
-		testpredicates.HasLabel("test-case", "hydration")); err != nil {
-		nt.T.Fatal(err)
-	}
+		testpredicates.HasLabel("test-case", "hydration")))
 
-	rs = getUpdatedRootSync(nt, configsync.RootSyncName, configsync.ControllerNamespace)
-	rsCommit, rsStatus, rsErrorSummary := getRootSyncCommitStatusErrorSummary(rs, nil, false)
+	// Validate nomos status
 	latestCommit := nt.RootRepos[configsync.RootSyncName].MustHash(nt.T)
-	expectedStatus := "SYNCED"
-	if err := validateRootSyncRepoState(latestCommit, rsCommit, expectedStatus, rsStatus, rsErrorSummary); err != nil {
-		nt.T.Fatal(err)
-	}
+	nt.Must(nt.Validate(configsync.RootSyncName, configsync.ControllerNamespace, &v1beta1.RootSync{},
+		testpredicates.RootSyncHasNomosStatus(latestCommit, "SYNCED")))
 
 	nt.T.Log("Make the hydration fail by checking in an invalid kustomization.yaml")
 	nt.Must(nt.RootRepos[configsync.RootSyncName].Copy("../testdata/hydration/resource-duplicate/kustomization.yaml", "./helm-overlay/kustomization.yaml"))
 	nt.Must(nt.RootRepos[configsync.RootSyncName].Copy("../testdata/hydration/resource-duplicate/namespace_tenant-a.yaml", "./helm-overlay/namespace_tenant-a.yaml"))
 	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Update kustomization.yaml with duplicated resources"))
+	latestCommit = nt.RootRepos[configsync.RootSyncName].MustHash(nt.T)
 	nt.Must(nt.Watcher.WatchObject(kinds.RootSyncV1Beta1(), configsync.RootSyncName, configsync.ControllerNamespace,
 		[]testpredicates.Predicate{
-			testpredicates.RootSyncHasRenderingError(status.ActionableHydrationErrorCode, ""),
+			testpredicates.RootSyncHasRenderingError(status.ActionableHydrationErrorCode, "failed to run kustomize build"),
```

Comment:
Why did the expected error message change here?

Reply:
Seems like it's just making the assertion stricter, which LGTM

Reply:
Similar text was in the validation Fatalf wrapper. So I moved it to the actual assertion so it would print what was expected when it fails.